presences/src/main/java/fr/openent/presences/security/AbsenceStatementsViewRight.java

```diff
@@ -13,16 +13,16 @@
 public class AbsenceStatementsViewRight implements ResourcesProvider {
     @Override
     public void authorize(HttpServerRequest request, Binding binding, UserInfos user, Handler<Boolean> handler) {
-        boolean result = WorkflowHelper.hasRight(user, WorkflowActions.MANAGE_ABSENCE_STATEMENTS.toString());
-        if (WorkflowHelper.hasRight(user, WorkflowActions.ABSENCE_STATEMENTS_VIEW.toString())) {
-            result = request.params().size() > 0;
-            List<String> student_ids = request.params().getAll("student_id");
-            result = student_ids.size() > 0;
-            for (String id : student_ids) {
+        boolean result = false;
+        if (WorkflowHelper.hasRight(user, WorkflowActions.MANAGE_ABSENCE_STATEMENTS.toString())) {
+            result = true;
+        } else if (WorkflowHelper.hasRight(user, WorkflowActions.ABSENCE_STATEMENTS_VIEW.toString())) {
+            List<String> studentIds = request.params().getAll("student_id");
+            result = request.params().size() > 0 && (!studentIds.isEmpty());
+            for (String id : studentIds) {
```

Comment:
result = request.params().size() > 0 && (!studentIds.isEmpty()) && studentsIds.stream().anyMatch((id) -> (user.getUserId().equals(id) || user.getChildrenIds().contains(id)))

 from 4 lignes to 1 ligne 😋

Reply:
et du coup 1000 fois plus simple, à partir de la ligne 18:
result = WorkflowHelper.hasRight(user, WorkflowActions.MANAGE_ABSENCE_STATEMENTS.toString()) ||
( WorkflowHelper.hasRight(user, WorkflowActions.ABSENCE_STATEMENTS_VIEW.toString()) &&
request.params().size() > 0 && (!studentIds.isEmpty()) && studentsIds.stream().anyMatch((id) -> (user.getUserId().equals(id) || user.getChildrenIds().contains(id)))
)